### PR TITLE
[improvement](vectorization) Use requires instead of specialization for doris::vectorized::Decimal

### PR DESCRIPTION
**Pull request by julic20s**
Use C++ 20 concept and requires to merge the implementation of `Decimal<T>` and `Decimal<wide::Int256>`. It improves the readability of Decimal and help the editor intellisense.
I removed the `Decimal<Int16>` in `be/src/vec/data_types/serde/data_type_decimal_serde.h`. It's can not be instantiation but appears in condition expressions

### CODE REVIEW

**Comment by julic20s** [comment]
run buildall

File: be/src/vec/core/types.h [hunk 1]
```diff
@@ -628,141 +676,19 @@ struct Decimal128I : public Decimal<Int128> {
...
-}
-inline bool operator>=(const Decimal256& x, const Decimal256& y) {
-    return x.value >= y.value;
-}
-inline bool operator==(const Decimal256& x, const Decimal256& y) {
-    return x.value == y.value;
-}
-inline bool operator!=(const Decimal256& x, const Decimal256& y) {
-    return x.value != y.value;
+inline std::strong_ordering operator<=>(const Decimal256& a, const Decimal256& b) {
```
Comment (yiguolei) [inline be/src/vec/core/types.h:684]: not use <=>， the  logic is not very clear.
And it is very dangerous.

**Comment by BiteTheDDDDt** [comment]
build failed

**Comment by julic20s** [comment]
run buildall

**Comment by julic20s** [comment]
run buildall

**Comment by julic20s** [comment]
Todo: 
- [ ] Remove Decimal<Int16> in `be/src/vec/data_types/serde/data_type_decimal_serde.h`. It is a illegal type.
- [ ] Use requires instead of static_assert in Decimal
- [ ] Do Regression tests for above

**Review by BiteTheDDDDt — DISMISSED** [review]

File: be/src/vec/core/types.h (hunk 1 above)
Comment (zclllyybb) [reply be/src/vec/core/types.h]: I think it's more clear than before?

**Comment by julic20s** [comment]
run buildall

**Comment by julic20s** [comment]
run buildall

**Comment by zclllyybb** [comment]
@jacktengg PTAL

**Review by jacktengg — COMMENTED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**